openedx/features/content_type_gating/partitions.py

```diff
@@ -93,8 +100,42 @@ def access_denied_message(self, block, user, user_group, allowed_groups):
         return None
 
     def _is_audit_enrollment(self, user, block):
+        """
+        Returns True if viewing course as:
+            1 - `Learner in Audit`
+            2 - `Specific student` and student has active audit enrollment
+        """
+        course_key = self._get_course_key_from_course_block(block)
+
+        if self._is_masquerading_as_generic_student(user, course_key):
+            return self._is_masquerading_audit_enrollment(user, course_key)
+        return self._has_active_enrollment_in_audit_mode(user, course_key)
+
+    def _is_masquerading_as_generic_student(self, user, course_key):
+        """
+        Returns True if viewing course as `Learner in {enrollment/group etc.}`
+        """
+        return (
+            is_masquerading_as_student(user, course_key) and
+            not is_masquerading_as_specific_student(user, course_key)
+        )
+
+    def _is_masquerading_audit_enrollment(self, user, course_key):
+        """
+        Returns True if viewing course as `Learner in Audit`
+        """
+        course_masquerade = get_course_masquerade(user, course_key)
+        if course_masquerade.user_partition_id == ENROLLMENT_TRACK_PARTITION_ID:
+            audit_mode_id = settings.COURSE_ENROLLMENT_MODES.get(CourseMode.AUDIT, {}).get('id')
+            return course_masquerade.group_id == audit_mode_id
+        return False
+
+    def _has_active_enrollment_in_audit_mode(self, user, course_key):
+        """
+        Returns True if user has active audit enrollment in course
```

Comment:
Lets make doc strings more descriptive. 

ie `Checks if the user has an audit and active enrollment in the provided course.`